File: sample-chat-java/app/src/main/java/com/quickblox/sample/chat/java/ui/widget/MessageTextViewCenter.java
package com.quickblox.sample.chat.java.ui.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.view.Gravity;
import android.widget.LinearLayout;

import com.quickblox.sample.chat.java.R;


public class MessageTextViewCenter extends MessageTextView {

    public MessageTextViewCenter(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void setLinearSide() {
        LayoutParams layoutParams = (LayoutParams) frameLinear.getLayoutParams();
        layoutParams.gravity = Gravity.CENTER;
        frameLinear.setLayoutParams(layoutParams);
    }

    @Override
    protected void setTextLayout() {
        viewTextStub.setLayoutResource(R.layout.widget_notification_msg_center);
        layoutStub = (LinearLayout) viewTextStub.inflate();
    }
}

File: sample-chat-kotlin/app/src/main/java/com/quickblox/sample/chat/kotlin/ui/views/MessageTextViewCenter.kt
package com.quickblox.sample.chat.kotlin.ui.views

import android.content.Context
import android.util.AttributeSet
import android.view.Gravity
import android.widget.FrameLayout
import android.widget.LinearLayout
import com.quickblox.sample.chat.kotlin.R

class MessageTextViewCenter(context: Context, attrs: AttributeSet) : MessageTextView(context, attrs) {

    override fun setLinearSide() {
        val layoutParams = frameLinear.layoutParams as FrameLayout.LayoutParams
        layoutParams.gravity = Gravity.CENTER
        frameLinear.layoutParams = layoutParams
    }

    override fun setTextLayout() {
        viewTextStub.layoutResource = R.layout.widget_notification_msg_center
        layoutStub = viewTextStub.inflate() as LinearLayout
    }
}